src/Microsoft.AspNetCore.Hosting/Internal/HostingApplication.cs

```diff
@@ -58,25 +64,52 @@ public void DisposeContext(Context context, Exception exception)
             if (exception == null)
             {
                 var diagnoticsEnabled = _diagnosticSource.IsEnabled("Microsoft.AspNetCore.Hosting.EndRequest");
-                var currentTimestamp = (diagnoticsEnabled || context.StartTimestamp != 0) ? Stopwatch.GetTimestamp() : 0;
+                var currentTimestamp = 0L;
+                var duration = 0L;
+                if (diagnoticsEnabled || context.StartTimestamp != 0)
+                {
+                    currentTimestamp = Stopwatch.GetTimestamp();
+                    duration = context.StartTimestamp - currentTimestamp;
```

Comment:
This breaks if EndRequest is enabled without BeginRequest.

if !_logger.IsEnabled && !_diagnosticSource.IsEnabled("Microsoft.AspNetCore.Hosting.BeginRequest") then context.StartTimestamp will == 0, causing duration to be negative.